contracts/JBETHPaymentTerminalStore.sol

```diff
@@ -442,42 +441,57 @@ contract JBETHPaymentTerminalStore {
     // Get a reference to the project's current funding cycle.
     fundingCycle = fundingCycleStore.currentOf(_projectId);
 
-    // Make sure the currencies match.
-    if (
-      _currency !=
-      directory.controllerOf(_projectId).overflowAllowanceCurrencyOf(
-        _projectId,
-        fundingCycle.configuration,
-        terminal
-      )
-    ) {
-      revert CURRENCY_MISMATCH();
-    }
-
     // Get a reference to the new used overflow allowance.
     uint256 _newUsedOverflowAllowanceOf = usedOverflowAllowanceOf[_projectId][
       fundingCycle.configuration
     ] + _amount;
 
     // There must be sufficient allowance available.
+    uint256 _allowanceOf = directory.controllerOf(_projectId).overflowAllowanceOf(
+        _projectId,
+        fundingCycle.configuration,
+        terminal
+      );
+
+    if(_newUsedOverflowAllowanceOf > _allowanceOf || _allowanceOf == 0) {
+      revert INADEQUATE_CONTROLLER_ALLOWANCE();
+    }
+
+    // Make sure the currencies match.
     if (
-      _newUsedOverflowAllowanceOf >
-      directory.controllerOf(_projectId).overflowAllowanceOf(
+      _currency !=
+      directory.controllerOf(_projectId).overflowAllowanceCurrencyOf(
         _projectId,
         fundingCycle.configuration,
         terminal
       )
     ) {
-      revert INADEQUATE_CONTROLLER_ALLOWANCE();
+      revert CURRENCY_MISMATCH();
     }
 
     // Convert the amount to wei.
     withdrawnAmount = (_currency == JBCurrencies.ETH)
       ? _amount
       : PRBMathUD60x18.div(_amount, prices.priceFor(_currency, JBCurrencies.ETH));
+    
+    // Get the current funding target
+    uint256 distributionLimit =
+      directory.controllerOf(_projectId).distributionLimitOf(
+        _projectId,
+        fundingCycle.configuration,
+        terminal
+      );
+
+    // Convert the target into wei, if needed
+    distributionLimit = _currency == JBCurrencies.ETH
```

Comment:
_currency here is the `overflowAllowanceCurrencyOf`. We need `distributionLimitCurrencyOf`.

Hopefully stack doesn't get too deep :|

Reply:
(yes, the currency for distributionLimit and overflowAllowance can be different)